packages/opentelemetry-instrumentation/src/autoLoaderUtils.ts

```diff
@@ -29,20 +30,28 @@ export function parseInstrumentationOptions(
 ): AutoLoaderResult {
   let instrumentations: Instrumentation[] = [];
   for (let i = 0, j = options.length; i < j; i++) {
-    const option = options[i] as any;
+    const option = options[i];
     if (Array.isArray(option)) {
       const results = parseInstrumentationOptions(option);
       instrumentations = instrumentations.concat(results.instrumentations);
-    } else if (typeof option === 'function') {
+    } else if (isInstrumentationClass(option)) {
       instrumentations.push(new option());
-    } else if ((option as Instrumentation).instrumentationName) {
+    } else if (isInstrumentation(option)) {
       instrumentations.push(option);
     }
   }
 
   return { instrumentations };
 }
 
+function isInstrumentationClass<T extends InstrumentationBase>(option: InstrumentationOption): option is new () => T {
```

Comment:
Nit: the function name should imho reflect what is happening inside (as close as possible). Because only type is checked here I would rather call it `isInstrumentationFunction` or even `isOptionAnInstrumentationFunction` WDYT ?

Reply:
It has `new` called on it after this guard so i think the existing name is fine.

Reply:
I think the correct name would be `isInstrumentationBaseSubclassConstructor` but too long!

Reply:
IMO the current name is fine.